.github/workflows/main.yaml

```diff
@@ -13,7 +13,31 @@ on:
   workflow_dispatch:
 
 jobs:
+
+  quality-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.2
+      - name: Install Poetry
+        run: |
+          pipx install poetry
+          pipx inject poetry poetry-dynamic-versioning
+      - uses: actions/setup-python@v5.6.0
+        with:
+          python-version: 3.13
+      - name: Check pyproject.toml and poetry.lock
+        run: poetry check
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Run codespell
+        run: tox -e codespell
+      - name: Run code format checks
+        run: tox -e format_check
```

Comment:
This just checks code formatting ATM. There are so many linting errors in this repo that it isn't worth preventing workflows from running due to failure until that situation is improved.